gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java

```diff
@@ -500,15 +504,26 @@ private void ensureDirectoryExists(FileSystem fs, Path path, Iterator<OwnerAndPe
 
       String group = ownerAndPermission.getGroup();
       String owner = ownerAndPermission.getOwner();
+      List<AclEntry> aclEntries = ownerAndPermission.getAclEntries();
       if (group != null || owner != null) {
         log.debug("Applying owner {} and group {} to path {}.", owner, group, path);
         fs.setOwner(path, owner, group);
       }
+      if (!aclEntries.isEmpty()) {
+        setAclOnPath(fs, path, aclEntries);
+      }
     } else {
       fs.mkdirs(path);
     }
   }
 
+  /*
+   * Creating an abstraction layer to support unit testing on raw local file system for validating if ACLs are set on a file path
+   */
+  protected void setAclOnPath(FileSystem theFs, Path path, List<AclEntry> aclEntries) throws IOException {
+    theFs.setAcl(path, aclEntries);
+  }
```

Comment:
Question, does LocalFileSystem in Hadoop support setAcl and a reader on acls? It could be more straightforward to test directly using set and get acls on a random file you write locally. 

If not this is fine

Reply:
Yes, I agree. I have extended LocalFileSystem to support setAcl and handle it accordingly for the purpose of unit testing in my latest commit. I no longer create this additional layer of abstraction.